pint/models/parameter.py

```diff
@@ -105,16 +113,16 @@ def units(self):
     def units(self, unt):
         # Check if this is the first time set units
         if hasattr(self, 'value'):
-            if self.units is None:
-                return
-            wmsg = 'Parameter '+self.name+' units has been reset to '+unt
-            log.warning(wmsg)
-            try:
-                if hasattr(self.value, 'unit'):
-                    temp = self.value.to(self.num_unit)
-            except:
-                log.warning('The value unit is not compatable with'
-                            ' parameter units,right now.')
+            if self.units is not None:
+                wmsg = 'Parameter '+self.name+' units has been reset to '+unt
+                wmsg += ' from '+self._units
+                log.warning(wmsg)
+                try:
+                    if hasattr(self.value, 'unit'):
+                        temp = self.value.to(self.num_unit)
```

Comment:
Instead of 'temp', you could also write '_'. Sometimes that is a clearer indication that the output won't be used.